src/providers/twitch/TwitchAccount.cpp

```diff
@@ -508,38 +507,50 @@ void TwitchAccount::loadEmoteSetData(std::shared_ptr<EmoteSet> emoteSet)
         return;
     }
 
-    NetworkRequest(Env::get().twitchEmoteSetResolverUrl.arg(emoteSet->key))
-        .cache()
-        .onSuccess([emoteSet](NetworkResult result) -> Outcome {
-            auto root = result.parseJson();
-            if (root.isEmpty())
+    getHelix()->getEmoteSetData(
+        emoteSet->key,
+        [emoteSet](HelixEmoteSetData emoteSetData) {
+            if (emoteSetData.ownerId.isEmpty() ||
+                emoteSetData.setId != emoteSet->key)
             {
-                return Failure;
+                qCWarning(chatterinoTwitch)
+                    << QString("Failed to fetch emoteSetData for %1, assuming "
+                               "Twitch is the onwer")
+                           .arg(emoteSet->key);
+
+                // most (if not all) emotes that fail to load are time limited event emotes owned by Twitch
+                emoteSet->channelName = "twitch";
+                emoteSet->text = "Twitch";
+
+                return;
             }
 
-            TwitchEmoteSetResolverResponse response(root);
-
-            auto name = response.channelName;
-            name.detach();
-            name[0] = name[0].toUpper();
-
-            emoteSet->text = name;
-            emoteSet->type = response.type;
-            emoteSet->channelName = response.channelName;
-
-            qCDebug(chatterinoTwitch)
-                << QString("Loaded twitch emote set data for %1")
-                       .arg(emoteSet->key);
-
-            return Success;
-        })
-        .onError([emoteSet](NetworkResult result) {
-            qCWarning(chatterinoTwitch)
-                << QString("Error code %1 while loading emote set data for %2")
-                       .arg(result.status())
-                       .arg(emoteSet->key);
-        })
-        .execute();
+            // emote set 0 = global emotes
+            if (emoteSetData.ownerId == "0")
+            {
+                //                emoteSet->channelName = QString();
```

Comment:
Did you intend to leave this commented out or left it by mistake?
```suggestion
```

Reply:
I'm not really sure if setting channelName in non-global emotesets from Twitch does not break things (puts them in Global category). I'll leave this commented out for now. Will probably remove it before finally merging into master.